### PR TITLE
Add support for pushing fabric structures

### LINKED ISSUE #232: GSA_Toolkit: Add support for pushing fabric structures

### PR DESCRIPTION
**Pull request by johannaisak**
Adds fabric material, fabric panel property, spacers, spacer properties, soap stress plus enables pushing of these

 ### NOTE: Depends on 

 ### Issues addressed by this PR

 Closes #232 

 

 ### Test files

https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/02_Pull%20Request/BHoM/GSA_Toolkit/%23232%20-%20AddSupportForPushingFabricStructures/fabric%20push.gh?csf=1&web=1&e=rBhXn8

 ### Changelog

 ### Additional comments

Adds fabric material, fabric panel property, spacers, spacer properties, soap stress + enables push of these

### CODE REVIEW

**Review by annastigenberg — APPROVED** [review]
Looks good to me!

**Comment by IsakNaslundBh** [comment]
@BHoMBot check project-compliance

**Comment by IsakNaslundBh** [comment]
@BHoMBot check project-compliance

**Comment by IsakNaslundBh** [comment]
@BHoMBot fix project file ref. 2718115227

File: GSA_Adapter/Convert/FromGsa/Elements/FEMesh.cs [hunk 1]
```diff
@@ -54,12 +54,12 @@ public static List<FEMesh> FromGsaFEMesh(IEnumerable<GsaElement> gsaElements, Di
...
                 int id = gsaMesh.Ref;
 
-                FEMeshFace face = new FEMeshFace() { NodeListIndices = Enumerable.Range(0, gsaMesh.NumTopo).ToList() };
+                FEMeshFace face = new FEMeshFace() { NodeListIndices = Enumerable.Range(0, gsaMesh.NumTopo).ToList(), OrientationAngle = gsaMesh.Beta * System.Math.PI / 180  };
                 face.SetAdapterId(typeof(GSAId), id);
 
                 FEMesh mesh = new FEMesh()
                 {
-                    Faces = new List<FEMeshFace>() { face },
+                    Faces = new List<FEMeshFace>() { new FEMeshFace() { NodeListIndices = Enumerable.Range(0, gsaMesh.NumTopo).ToList() } },
```
Comment (IsakNaslundBh) [inline GSA_Adapter/Convert/FromGsa/Elements/FEMesh.cs:62]:
```suggestion
                    Faces = new List<FEMeshFace>() { face },
```

**Review by IsakNaslundBh — CHANGES_REQUESTED** [review]
Small tweak below. Should be using the constructed face, rather than pick a new one

**Comment by IsakNaslundBh** [comment]
@BHoMBot check installer

**Review by IsakNaslundBh — APPROVED** [review]
Works fine. Nice work!

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
Merging per agreement with @IsakNaslundBh following updates to required checks from `check-installer` to `installer` (and needing to include versioning).

image

**Pull request merged**